### PR TITLE
chore: Rename SD-Core charms by adding -k8s as a suffix

### PR DESCRIPTION
**Pull request by gatici**
# Description

Add -k8s as a suffix in both github and Charmhub name to follow [TE042](https://docs.google.com/document/d/1R0UzoPr3vvorhbBJYBz-5gZkTCTL2a9_R6xV8K4_i-8/edit) . 

- Fetch new fiveg_nrf library
- Pin macaroonbakery to 1.3.2 which installed by juju client indirectly through pytest_operator.
      - The next protobuf version 4.21.0 has the breaking changes with the existing code
      - https://protobuf.dev/news/2022-05-06/#python-updates

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -37,8 +37,8 @@
 CERTIFICATE_COMMON_NAME = "nssf.sdcore"
 
 
-class NSSFOperatorCharm(CharmBase):
-    """Main class to describe juju event handling for the SD-Core NSSF operator."""
+class NSSFK8sOperatorCharm(CharmBase):
```
Comment (Gmerold) [inline src/charm.py:40]: I think there's no need to change the class name.

File: README.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
-# SD-Core NSSF Operator (k8s)
-[![CharmHub Badge](https://charmhub.io/sdcore-nssf/badge.svg)](https://charmhub.io/sdcore-nssf)
+# SD-Core NSSF K8s Operator
+[![CharmHub Badge](https://charmhub.io/sdcore-nssf-k8s/badge.svg)](https://charmhub.io/sdcore-nssf-k8s)
 
-Charmed Operator for the SD-Core Network Slice Selection Function (NSSF).
+Charmed K8s Operator for the SD-Core Network Slice Selection Function (NSSF).
```
Comment (Gmerold) [inline README.md:4]: `Charmed K8s Operator` to me, sounds a bit like it was meant only for Charmed K8s. If it's just my opinion, feel free to ignore this ;)

File: README.md [hunk 3]
```diff
@@ -1,18 +1,18 @@
-# SD-Core NSSF Operator (k8s)
-[![CharmHub Badge](https://charmhub.io/sdcore-nssf/badge.svg)](https://charmhub.io/sdcore-nssf)
+# SD-Core NSSF K8s Operator
```
Comment (Gmerold) [inline README.md:1]:
```suggestion
# SD-Core NSSF Operator for K8s
```

File: src/charm.py [hunk 4]
```diff
@@ -2,7 +2,7 @@
 # Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-"""Charmed operator for the SD-Core NSSF service."""
+"""Charmed operator for the SD-Core NSSF K8s service."""
```
Comment (Gmerold) [inline src/charm.py:5]:
```suggestion
"""Charmed operator for the SD-Core NSSF for K8s."""
```

**Review by Gmerold — CHANGES_REQUESTED** [review]

**Review by Gmerold — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (gatici) [reply README.md]: Done

File: src/charm.py (hunk 4 above)
Comment (gatici) [reply src/charm.py]: Done

File: src/charm.py (hunk 1 above)
Comment (gatici) [reply src/charm.py]: It kept as same.

**Pull request merged**